.github/workflows/yki-frontend.yml

```diff
@@ -26,15 +27,38 @@ jobs:
           node-version: lts/*
           # Note: does not cache node_modules
           # Use instead actions/cache explicitly if caching node_modules is desired
-          cache: 'npm'
+          cache: "npm"
           cache-dependency-path: src/main/js/
 
       - name: Install NPM dependencies
         run: npm ci --prefix src/main/js/
 
-      - name: Run tests
+      - name: Run Jest tests
         run: npm test --prefix src/main/js/
 
+      - name: Cache Cypress
+        uses: actions/cache@v3
+        id: cypress-cache
+        with:
+          path: "~/.cache/Cypress"
+          key: ${{ runner.os }}-cypress-cache-${{ hashFiles('src/main/js/package.json') }}
+      - name: Install Cypress
+        uses: cypress-io/github-action@v4
+        with:
+          working-directory: ${{ env.FRONTEND_DIR }}
+          runTests: false
+        if: steps.cypress-cache.outputs.cache-hit != 'true'
+      - name: Run Cypress tests
+        uses: cypress-io/github-action@v4
+        with:
+          working-directory: ${{ env.FRONTEND_DIR }}
+          install: false
+          build: npm run build:ci
+          start: npm run start:ci
+          wait-on: "http://localhost:3000"
+          command: npm run cypress:run
```

Comment:
Videon tallentamisen voisi poistaa tästä. `https://docs.cypress.io/guides/guides/screenshots-and-videos#Videos`